sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpReceiver.cs

```diff
@@ -420,6 +420,34 @@ private async Task<IReadOnlyList<ServiceBusReceivedMessage>> ReceiveMessagesAsyn
             }
         }
 
+        /// <summary>
+        /// Drains an AMQP link. When drain failure happens we make sure to close the link to ensure ordering.
+        /// </summary>
+        /// <param name="link">The AMPQ link to drain.</param>
+        /// <param name="cancellationToken">An optional <see cref="CancellationToken"/> instance to signal the request to cancel the operation.</param>
+        /// <returns>A task to be resolved on when the operation has completed.</returns>
+        public async Task DrainLinkAsync(ReceivingAmqpLink link, CancellationToken cancellationToken)
```

Comment:
```suggestion
        public async Task SafeDrainLinkAsync(ReceivingAmqpLink link, CancellationToken cancellationToken)
```

Since it came up in conversation - it may be a good idea to apply the "Safe" naming pattern here, since we explicitly swallow errors.